### ISSUE TITLE
A Back Button

### ISSUE THREAD

**Issue by AqilCont** [ppy/osu#5063]
**Describe the new feature:** 
There should be a back button on the top when clicking links on the website browser on the Lazer client. I clicked on WalkingTuna's top play and it took me to here but when I wanted to go back to WalkingTuna's profile, I had to take the long process of finding his profile again.
![osu_2019-06-19_09-27-09](https://user-images.githubusercontent.com/32044067/59735790-c1fdad00-9274-11e9-87a3-1d5c66736234.jpg)

**Proposal designs of the feature:** 
You can add a back button or something like that at the blank gray space you have at the top of the web browser(which serves no obvious purpose and just exists for some reason). This will probably shorten the time people have to take to look through stuff on the Lazer client.

**Comment by smoogipoo** [comment]
Probably should happen alongside #4886

**Comment by AqilCont** [comment]
~~wow that was extremely fast~~

**Comment by peppy** [comment]
Not sure if it should be implemented via the #4886 button; probably a separate set of controls to avoid confusion.

**Comment by AqilCont** [comment]
Oh yeah, I just noticed. If you look in the picture at the beatmap's tags, they almost touch the rankings panel. You should probably fix that too.

**Comment by Joehuu** [comment]
Please post different issues for those, and make sure to search first to avoid duplicates.

**Comment by Bryceed** [comment]
> Edge click in profile's beatmap info panel back to menu, instead to return one-level navigation #12514

@frenzibyte @peppy 
That's not a implementation of a new function, is the same thing working in different ways in osu!direct panel and user profile which can disrupt the users navigation if they unintentionally clicks in beatmap card intead of the download button (thing that can be reversed in osu!direct).

**Comment by peppy** [comment]
It's the same area of UX that needs improving. The way it's done on the beatmap listing can be thought of as a temporary best-effort (we intentionally don't close the listing when a beatmap is selected). The same can't easily be implemented for other scenarios. Basically, we need a better method of handling back/forward navigation, which is what this issue is tracking.

**Comment by akippnn** [comment]
It's been a while since this issue had an update and I was directed here from #17577.

The simplest way is to just add the back menu in the overlay itself before the title of the overlay. Like this:
![unknown](https://user-images.githubusercontent.com/80670461/199616938-69522965-1124-45b9-9ec7-d5c5d50bdaff.png)
I couldn't use the fonts lazer used so I had to take a screenshot from the in-game client.

The only problem is that we have to navigate up to be able to press this button or visually tell people "hey, you can go back to the previous overlay if you want." So an example I could think of:
![unknown(1)](https://user-images.githubusercontent.com/80670461/199617272-3df879a3-73c4-47e5-a7b2-c426217f5bd8.png)
Not that I really want to see "player info" anchored since that is personally redundant info and I would rather see the player username there, but this is just for demonstration.

If the button does contract like above, hover interaction would tell the player what they're navigating back to.
![unknown(2)](https://user-images.githubusercontent.com/80670461/199617682-d3329c69-821c-47b9-a1e0-b1db0c906d31.png)
Preferably not just telling us we're going back to a beatmap info though, but again just demonstration purposes for now since I couldn't load fonts in Figma.

There are other little things I thought would be interesting to see:
- Gestures designed primarily for touch devices, where dragging from the utmost left area to right on an extent would trigger going back to the previous overlay (with animation as an indication this is being done).
- If it is reactive to mouse side buttons used primarily for navigation. Pressing M4 on an overlay should prevent it from triggering the global "back" keybind.

**Comment by peppy** [comment]
Curious if @arflyte has considered this in newer UI designs.

**Comment by akippnn** [comment]
> Curious if @arflyte has considered this in newer UI designs.

Would be quite a change presumably, considering there's now another content box to anchor. It seems like a good change though but I think it warrants its issue (while linking this one). Here's sort of a pre-polish comparison preview:
![image](https://user-images.githubusercontent.com/80670461/199952561-09edd14d-1cba-4f58-a529-cb6f19121363.png)
The anchored box is necessary to keep the back button anchored. On the other hand, it doesn't need all of that actually, but a floating button feels a bit jarring to me personally, making overlays feel like a webpage being loaded.

**Comment by peppy** [comment]
I think that could work quite well with the last design.

**Comment by Ale1sterCrowley** [comment]
how do I upvote this issue

![image](https://github.com/ppy/osu/assets/127149617/56bcecfe-3098-46c5-99a5-79649658bb0f)

**Comment by TheAirBlow** [comment]
> how do I upvote this issue
> 
> ![image](https://private-user-images.githubusercontent.com/127149617/291736688-56bcecfe-3098-46c5-99a5-79649658bb0f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMDQ4NjYsIm5iZiI6MTcxMDMwNDU2NiwicGF0aCI6Ii8xMjcxNDk2MTcvMjkxNzM2Njg4LTU2YmNlY2ZlLTMwOTgtNDZjNS05OWE1LTc5NjQ5NjU4YmIwZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxM1QwNDM2MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NjAzNDVlNTA4M2EyNjJkZTFkZGRlMTAyZjFlOTFmZjEyYzdiZDE2ZTcwZjA1MGNjNzg4ZTJiM2U0NTU4NjIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.FwgGLAoYToblYLp7a9oDoU5Y67cZmAFI-J1DYDnnGks)

This is an issue, you can't. Just leave a like, and comment if the issue stays inactive for too long (e.g. in a week at least to not be annoying)

**Comment by TheAirBlow** [comment]
> Please don't comment to remove inactivity, the devs are aware about everything and are putting their focus on gameplay breaking issues, sometimes not requiring a discussion over. Bump-type comments just flood their inboxes/notifications rather than being helpful.

I agree, but people do sometimes forget... Once a few weeks should be fine imo (just because that is true for me xD)

If devs in this repo organize properly, it shouldn't be necessary, which looks to be true.

**Comment by frenzibyte** [comment]
Please don't, not even in a few weeks. The repository is organised quite properly, with priority labels given to most issue threads according to how important they are in the grand scheme of things. "Forget" is not the right word, we cannot find the bandwidth necessary to work on this issue thread specifically, we're not "forgetting" to implement it.

**Comment by moonfight1** [comment]
what about it just being alt+arrow or backward and forward on mice like it is in browsers

**Comment by Cybernetics354** [comment]
i'm sorry but is there any updates on this issue? i found it exhausting to redownload my played beatmaps on new device (click profile -> loading -> click beatmap -> download -> close the panel -> click profile -> loading -> ...)  

if there's a back button then i could just click profile -> loading -> click beatmap -> download -> back -> click beatmap -> download -> ...

**Comment by akippnn** [comment]
> i'm sorry but is there any updates on this issue? i found it exhausting to redownload my played beatmaps on new device (click profile -> loading -> click beatmap -> download -> close the panel -> click profile -> loading -> ...)
> 
> if there's a back button then i could just click profile -> loading -> click beatmap -> download -> back -> click beatmap -> download -> ...

I agree. Though I also agree it's not P0, but it definitely needs attention.

I'll try to implement this on my own at some point. I also have some other ideas with the overlay UX besides the back button, but I'll separate those concerns I think.
